eessi/testsuite/hooks.py

```diff
@@ -180,25 +180,28 @@ def filter_valid_systems_by_device_type(test: rfm.RegressionTest, required_devic
     Filter valid_systems by required device type and by whether the module supports CUDA,
     unless valid_systems is specified with --setvar valid_systems=<comma-separated-list>.
     """
-    if not test.valid_systems:
-        is_cuda_module = is_cuda_required_module(test.module_name)
-        valid_systems = ''
+    if test.valid_systems:
+        # valid_systems is specified, so don't filter
+        return
+
+    is_cuda_module = is_cuda_required_module(test.module_name)
 
-        if is_cuda_module and required_device_type == DEVICES['GPU']:
-            # CUDA modules and when using a GPU require partitions with 'gpu' feature
-            valid_systems = f'+{FEATURES["GPU"]}'
+    if is_cuda_module and required_device_type == DEVICES['GPU']:
+        # CUDA modules and when using a GPU require partitions with 'gpu' feature and
+        # 'gpu_vendor=nvidia' extras
+        valid_systems = f'+{FEATURES["GPU"]} %gpu_vendor={GPU_VENDORS["NVIDIA"]}'
```

Comment:
use `GPU_VENDOR` constant:

```suggestion
        valid_systems = f'+{FEATURES["GPU"]} %{GPU_VENDOR}={GPU_VENDORS["NVIDIA"]}'